frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx

```diff
@@ -0,0 +1,274 @@
+      collection: ROOT_COLLECTION,
+    });
+    await setup({
+      collections: COLLECTIONS,
+      dashboard: dashboardInPublicCollection,
+    });
+
+    assertBreadcrumbs([ROOT_COLLECTION]);
+
+    expect(
+      screen.getByRole("menuitem", {
+        name: COLLECTION.name,
+      }),
+    ).toBeInTheDocument();
+
+    expect(
+      screen.queryByRole("menuitem", {
+        name: PERSONAL_COLLECTION.name,
+      }),
+    ).not.toBeInTheDocument();
+  });
+
+  it("should show all questions when adding questions to dashboards in the root collection (public collection)", async () => {
+    const dashboardInRootCollection = createMockDashboard({
+      collection: ROOT_COLLECTION,
+    });
+
+    const collectionItems = [
+      createMockCollectionItem({
+        id: getNextId(),
+        model: "card",
+        name: "question 1",
+      }),
+      createMockCollectionItem({
+        id: getNextId(),
+        model: "card",
+        name: "question 2",
+      }),
+    ];
+    await setup({
+      collections: COLLECTIONS,
+      dashboard: dashboardInRootCollection,
+      collectionItems,
+    });
+
+    assertBreadcrumbs([ROOT_COLLECTION]);
+
+    collectionItems.forEach(collectionItem => {
+      expect(
+        screen.getByRole("menuitem", {
+          name: collectionItem.name,
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  it("should show all questions when adding questions to dashboards in public collections", async () => {
+    const dashboardInPublicSubcollection = createMockDashboard({
+      collection: SUBCOLLECTION,
+    });
+
+    const collectionItems = [
+      createMockCollectionItem({
+        id: getNextId(),
+        model: "card",
+        name: "question 1",
+      }),
+      createMockCollectionItem({
+        id: getNextId(),
+        model: "card",
+        name: "question 2",
+      }),
+    ];
+    await setup({
+      collections: COLLECTIONS,
+      dashboard: dashboardInPublicSubcollection,
+      collectionItems,
+    });
+
+    assertBreadcrumbs([ROOT_COLLECTION, COLLECTION, SUBCOLLECTION]);
+
+    collectionItems.forEach(collectionItem => {
+      expect(
+        screen.getByRole("menuitem", {
+          name: collectionItem.name,
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  it("should show all collections when adding questions to dashboards in personal subcollections", async () => {
+    const dashboardInPersonalSubcollection = createMockDashboard({
+      collection: PERSONAL_SUBCOLLECTION,
+    });
+    await setup({
+      collections: COLLECTIONS,
+      dashboard: dashboardInPersonalSubcollection,
+    });
+
+    assertBreadcrumbs([
+      ROOT_COLLECTION,
+      PERSONAL_COLLECTION,
+      PERSONAL_SUBCOLLECTION,
+    ]);
+
+    expect(screen.getByText("Nothing here")).toBeInTheDocument();
+  });
+
+  it("should show all questions when adding questions to dashboards in personal subcollections", async () => {
```

Comment:
test Dashboards > #2 from [the epic](https://github.com/metabase/metabase/issues/34733)